test/philomena_web/controllers/deactivation_controller_test.exs

```diff
@@ -24,8 +26,10 @@ defmodule PhilomenaWeb.DeactivationControllerTest do
       assert Memory.all()
              |> Enum.find(&(&1.subject == "Reactivation instructions for your account")) != nil
 
-      user = Users.get_user!(user.id)
-      assert user.deleted_by_user_id == user.id
+      assert_retry(fn ->
```

Comment:
Before doing this I would want to find out why the update is not persisted by this point

Reply:
My immediate guess is that sending a `POST` request to that internal API doesn't cause the thread to wait for the actual "side-effects" of that to be finished?

Often we get the correct version:
1. POST request to `localhost/reactivations`
a. SQL update / transaction is sent.
2. Page is returned.
1. a. Is written to database.
3. Test attempts to check database.
4. Expected value.

Sometimes we get:
1. POST request to `localhost/reactivations`
a. SQL update / transaction is sent.
2. Page is returned.
3. Test attempts to check database.
4. Erroneous value.
1. a. Is written to database.

Reply:
Repo.update is supposed to ensure the update is either fully committed or fully rolled back when the function call returns. If the conn's render directly varies based on the result of that Repo.update, then there shouldn't be any race. It would only race if the deactivation got kicked to a background job.

Reply:
Seems to be resolved, current theory is background job was coming from testing swoosh's memory.